### ISSUE TITLE
Weird behaviour when marking text

### ISSUE THREAD

**Issue by yotamN** [johnfactotum/foliate#402]
When I select a word with a double click it works the first time but when I try to do it again when I already have a word marked the program mark the whole paragraph instead of only the new word.

I'm not sure if this behaviour is on purpose but it's different from the operating system and a bit confusing.

**Comment by digitalethics** [comment]
I agree. I think this happens if your subsequent second double click is shortly after the first one. Maybe some changes can be made to the time fuse between first and subsequent click events?

**Comment by digitalethics** [comment]
@yotamN You could specify the title to something like "Time fuse too short between click events when highlighting text".

**Comment by johnfactotum** [comment]
I think it's a quirk in how GTK 3 handles popovers. This doesn't happen in GTK 4. In GTK 4, double clicking when the popover is active will not select any text, which is expected because the popover is modal, i.e. the first click closes the popover, and only the second click is registered on the underlying widget. Though arguably it would be more ideal if the popover isn't modal.
